### ISSUE TITLE
Tree Dock - Icon Visibility Toggling

### ISSUE THREAD

**Issue by wingedadventurer** [godotengine/godot-proposals#1608]
**Describe the project you are working on:**
Any project.

**Describe the problem or limitation you are having in your project:**
Tree Dock can get cluttered with all of the icons visible, as seen in example below:

![old](https://user-images.githubusercontent.com/26509794/95013652-fea81000-0641-11eb-8e2f-b20525cf15f8.png)

There are some icons that i don't care about. For example:

* I don't care that KinematicBody2D nodes don't have collision shape, I intend to add them later.
* I also don't care that a node is in a group.

Finally, when i worked on 1366x768 resolution before, the tree dock would get very cluttered, especially when working with UI for example. Ability to remove some icons could help with this issue a bit.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Allow hiding icons by their type. User can easily hide the icons they don't care about.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
The mockups below illustrate how it looks with some icons hidden:

![new_1](https://user-images.githubusercontent.com/26509794/95013765-ade4e700-0642-11eb-9080-6d39afc4aa49.png)

This is the View popup mockup:

![new_2](https://user-images.githubusercontent.com/26509794/95013775-bf2df380-0642-11eb-80c8-6c2f365a286d.png)

These are just mockups, so things like order of icons in the popup, or `View` button alternative, are subject to change.

Any additional ideas are welcome. :)

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
N/A (editor enhancement)

**Is there a reason why this should be core and not an add-on in the asset library?:**
2D and 3D editors already provide ways to toggle visibility of some elements, why not have something similar for the tree dock as well?

**Comment by Calinou** [comment]
The top bar's "View" menu might make the search field very short if your dock is narrow.

**Comment by wingedadventurer** [comment]
@Calinou Good point. Perhaps the search field can be moved in its own line below? Like this:

![new_new_search](https://user-images.githubusercontent.com/26509794/95021792-80fcf800-0673-11eb-996f-7b9497ec1477.png)

This would also make it more consistent with `Inspector` dock:

![10-04-2020_06-56-12 PM](https://user-images.githubusercontent.com/26509794/95021766-498e4b80-0673-11eb-920c-1f42842fac1a.png)

**Comment by Calinou** [comment]
@wingedadventurer That design wastes a lot of vertical space. This vertical space is precious on small laptops with a 16:9 display :slightly_smiling_face:

**Comment by volzhs** [comment]
![Screenshot_20201005_022428](https://user-images.githubusercontent.com/8281454/95022380-f687b900-06b1-11eb-9efc-8af8fb46288d.png)
![Screenshot_20201005_022437](https://user-images.githubusercontent.com/8281454/95022409-1dde8600-06b2-11eb-831a-7129707a1d23.png)
here's how blender handles this.
(though I didn't know these options are in there before searching on the internet)
